cli/cmd/install.go

```diff
@@ -443,7 +450,11 @@ var installCmd = &cobra.Command{
 			ControllerComponentLabel: k8s.ControllerComponentLabel,
 			CreatedByAnnotation:      k8s.CreatedByAnnotation,
 		}
-		return render(config, os.Stdout)
+		err := render(config, buf)
+		if err != nil {
+			return err
+		}
+		return k8sController.InjectYAML(buf, os.Stdout, k8sController.DefaultProxyConfig)
```

Comment:
more of a product question: should we do the injection via the `install` command, or keep it in `inject`?

i really liked the `conduit install | conduit inject` pattern. it's clearer to the user what's going on, and gets them into the habit of running the inject command. it also has the added benefit that if the user is setting any params on their other `conduit inject` commands, they can follow the same pattern with installing the control plane:

```bash
cat app.yml | conduit inject - --image-pull-policy Always
conduit install | conduit inject - --image-pull-policy Always
```

as written, every param we support via `inject` will have to be supported via `install` as passthrough:

```bash
cat app.yml | conduit inject - --image-pull-policy Always
conduit install --image-pull-policy Always
```

Reply:
/cc @briansmith would like to here your thoughts on this

Reply:
> i really liked the conduit install | conduit inject pattern.

I like it too and I think the *implementation* should work like that.

> it's clearer to the user what's going on

I don't think this is true. First, I don't think users care too much whether the control plane is implemented in terms of the proxy or not, and we should strive to make it so they don't have to care. Secondly, I think they'll wonder "what happens if I don't do the `conduit inject` step during installation?" Once we start depending on the proxy being used in the controlller, trying to skip the `conduit inject` step would be fatal.

> gets them into the habit of running the inject command

I really see `conduit inject` is a temporary workaround. We need to support a more automated mechanism where people don't need to manually inject the proxy into their applications.

> as written, every param we support via inject will have to be supported via install as passthrough

That's the worst-case scenerio. Even in the worst case, since `conduit inject` and `conduit install` are both in the `conduit` program, they can easily share command line flag handling. In fact, we already have such sharing today.

I'm not sure every `conduit inject` parameter needs to be supported by `conduit install`. You mentioned the control for the imagePullPolicy, but I think `conduit install` already needs (has?) a control for imagePullPolicy for the controller images. What other inject flags would actually need to be supported in `conduit install`?

Anyway, it is likely that, at least in initial configurations, we'll need to tweak the details of how the proxy is used in the control plane, e.g. by skipping ports and/or by configuring TLS in an unusual way. The `conduit inject` step would get quite complicated and every release potentially would have a different one.

Even if the `conduit inject` step were dead simple with no flags needed, I'd still prefer to have `conduit install` do the injection automatically.